lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -266,15 +313,272 @@ def mock_run(args, **kwargs):
+        zero, it is handled successfully."""
+        source_tarball, source_md5, md5 = tarball
+        tarball_name = source_tarball.name.removesuffix(".tar.xz")
+
+        def mock_run(args, **kwargs):
+            if args[0] == "tar":
+                assert args == [
+                    "tar",
+                    "-xf",
+                    "--no-same-owner",
+                    "--delay-directory-restore",
+                    "--force-local",
+                    (server_config.ARCHIVE / "ABC" / source_tarball.name),
+                ]
+                return subprocess.CompletedProcess(
+                    args, returncode=0, stdout="Successfully Unpacked!", stderr=None
+                )
+            else:
+                assert args[0] == "find"
+                return subprocess.CompletedProcess(
+                    args,
+                    returncode=1,
+                    stdout=None,
+                    stderr="Some error unpacking tarball\n",
+                )
+
+        monkeypatch.setattr("subprocess.run", mock_run)
+        tree = FileTree(server_config, make_logger)
+        tree.create("ABC", source_tarball)
+
+        tarball = tree.find_dataset(md5)
+        assert tarball.resource_id == md5
+        assert tarball.resource_id in tree
+        with pytest.raises(TarballModeChangeError) as exc:
+            tree.unpack(md5)
+        assert (
+            str(exc.value)
+            == f"An error occurred while changing file permissions of {tarball_name}: tar exited with status 1:  'Some error unpacking tarball'"
+        )
+
+    def test_unpack_move_src(
+        self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
+    ):
+        """Show that, when unpack directory for moving results is not present and
+        raises FileNotFoundError Exception, it is handled successfully."""
+        source_tarball, source_md5, md5 = tarball
+        tarball_name = source_tarball.name.removesuffix(".tar.xz")
+
+        def mock_move(src: str, dest: str):
+            raise FileNotFoundError(f"No such file or directory: '{src}'")
+
+        def mock_run(args, **kwargs):
+            if args[0] == "tar":
+                assert args == [
+                    "tar",
+                    "-xf",
+                    "--no-same-owner",
+                    "--delay-directory-restore",
+                    "--force-local",
+                    (server_config.ARCHIVE / "ABC" / source_tarball.name),
+                ]
+                return subprocess.CompletedProcess(
+                    args, returncode=0, stdout="Successfully Unpacked!", stderr=None
+                )
+            else:
+                assert args[0] == "find"
+                return subprocess.CompletedProcess(
+                    args,
+                    returncode=0,
+                    stdout="Successfully Changed File Permissions!",
+                    stderr=None,
+                )
```

Comment:
For this test, I would go with something like this or, possibly, even use the `lambda` equivalent:
```
        def mock_run(args, **kwargs):
            return subprocess.CompletedProcess(args, returncode=0, stdout="Success!", stderr=None)
```
We don't need to do anything differently for `tar` vs. `find`, and we don't need to assert anything (because we've done it in the prior scenarios), so we don't need the conditional if we just return the same `stdout` message (which we ultimately ignore anyway).

Ditto line 467, 516, et al.